torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -1258,54 +1540,37 @@ def apply(self, fn: Callable[[nn.Module], None]) -> "FullyShardedDataParallel":
 
         return ret
 
-    # setting two factors 'self.gradient_predivide_factor'
-    # and 'self.gradient_postdivide_factor' to avoid underflow and overflow
     def _get_gradient_predivide_factor(self, world_size: int) -> float:
+        """
+        Returns a pre-divide factor to prevent underflow/overflow during
+        gradient synchronization. The post-divide factor should be the
+        reciprocal of the pre-divide factor.
```

Comment:
TODO: This statement is not correct. The post-divide factor is actually the world size divided by the pre-divide factor.